### PR TITLE
refactored mysql versioned tests to rely on promises to bootstrap db

### LINKED ISSUE #52: Update tap to version 15
Attempt to update tap to version 15.0.9 caused test failures because of extensive use of `done` in before/after test setups. `done` is called throughout the setup process. We suggest converting the setup to something more promise based and compatible with the tap setup and get rid of the `async` usage. 

Here's an example of the done being used as a callback in setup:
https://github.com/newrelic/node-newrelic-mysql/blob/main/tests/versioned/common/setup.js#L20-L42

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Upgraded tap to v15. 

## Links
Closes #52 

## Details
The refactor required restructuring some of the test suites to have a beforeEach instead of 1 inline test.  This was not required but made it easier to await the setup in a before each and wrap the 1 test in a child `t.test`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,13 +31,13 @@
   },
   "devDependencies": {
     "@newrelic/test-utilities": "^5.1.0",
-    "async": "^2.6.1",
+    "async": "^3.2.0",
```
Comment (bizob2828) [inline package.json:34]: had to upgrade to allow for async handlers within `async.*`

File: tests/versioned/common/pooling.js [hunk 2]
```diff
@@ -11,18 +11,26 @@ const utils = require('@newrelic/test-utilities')
 
 module.exports = (t, requireMySQL) => {
   t.test('MySQL instrumentation with a connection pool', {timeout: 30000}, (t) => {
```
Comment (bizob2828) [inline tests/versioned/common/pooling.js:13]: refactored this to make it easier to await setup and not deal with wrapping the rest in an async function to properly handling the nested suites

File: tests/versioned/common/setup.js [hunk 3]
```diff
@@ -17,74 +17,68 @@ const params = exports.params = {
   database: 'agent_integration_' + Math.floor(Math.random() * 1000)
 }
 
-function setup(mysql, cb) {
```
Comment (bizob2828) [inline tests/versioned/common/setup.js:20]: cleaned this up a bit to just await 2 diff async functions `executeDb` and `setupDb`. My head was spinning with the nested `async.series` and `async.eachSeries`

File: tests/versioned/common/transactions.js [hunk 4]
```diff
@@ -14,20 +14,23 @@ const params = setup.params
 
 module.exports = (t, requireMySQL) => {
   t.test('MySQL transactions', {timeout: 30000}, (t) => {
```
Comment (bizob2828) [inline tests/versioned/common/transactions.js:16]: refactored this to make it easier to await setup and not deal with wrapping the rest in an async function to properly handling the nested suites

File: tests/versioned/common/pooling.js [hunk 5]
```diff
@@ -76,15 +84,15 @@ module.exports = (t, requireMySQL) => {
...
-      t.error(err)
+
+    t.test('basic transaction', (t) => {
       t.notOk(helper.getTransaction(), 'no transaction should be in play yet')
       helper.runInTransaction((txn) => {
         var context = {
           id: 1
         }
         dal.lookup(context, (error, row) => {
-          if (!t.error(err) || !t.transaction(txn)) {
```
Comment (carlo-808) [inline tests/versioned/common/pooling.js:87]: 👀  😄

**Review by carlo-808 — APPROVED** [review]
Nice refactor with the async db setup. 👍

**Pull request merged**